### PR TITLE
Slice stream improvements and fixes

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: src/main/java/io/airlift/slice/Slice.java [hunk 1]
```diff
@@ -647,27 +647,25 @@ public void setBytes(int index, byte[] source, int sourceIndex, int length)
...
-        int remaining = length;
-        while (remaining > 0) {
-            int bytesRead = in.read(bytes, 0, Math.min(bytes.length, remaining));
+
+        while (length > 0) {
+            int bytesRead = in.read(bytes, 0, Math.min(bytes.length, length));
             if (bytesRead < 0) {
-                // if we didn't read anything return -1
-                if (remaining == length) {
-                    return -1;
```
Comment (martint) [inline src/main/java/io/airlift/slice/Slice.java]: If the byte is outside the 0-127 range this won't return valid ascii

Comment (dain) [inline src/main/java/io/airlift/slice/Slice.java]: Did you comment on the wrong line?

Comment (martint) [inline src/main/java/io/airlift/slice/Slice.java]: Yeah, that should've gone in `toStringAscii`

**Comment by martint** [comment]
looks good

**Pull request merged**